### DISCUSSION TITLE
Plotly figures lose data when switching between tabs

### DISCUSSION THREAD

**Discussion by Scheeer** [zauberzeug/nicegui#845] [Q&A] [answered]
### Description

when periodicaly updating ui.plot with an async method, the plotted data is reset to the original values when swapping between tabs  
this seems similar to #488 

minimum working example:
```python
from nicegui import ui
import plotly.graph_objects as go
import numpy as np
import asyncio

fig = go.Figure()

async def async_task():
    while connect_switch.value:
        await asyncio.sleep(5)
        fig.data = []
        fig.add_trace(go.Scatter(x=np.arange(100), y=np.random.random(100),text="100 Random points"))
        plot1.update()
with ui.tabs() as tabs:
    ui.tab('A')
    ui.tab('B')
with ui.tab_panels(tabs, value='A'):
    with ui.tab_panel('A'):
        plot1 = ui.plotly(fig).classes('w-full h-25')
        fig.add_trace(go.Scatter(x=np.arange(100), y=np.zeros(100),text="First data set"))
        plot1.update()
        print("tab")
    with ui.tab_panel('B'):
        ui.label('Flipping here will set plot data back to zeros')
        connect_switch = ui.switch('Start async plotting',on_change=async_task)

ui.run(port=8081)
```

We create a ui.plotly element that gets updated every 5 seconds with new random data. 
We would expect this data to be shown even when switching between the two tabs.
Instead we see the first set of data of all zeros after we swapped to another tab and back.

**Comment by rodja** [comment, marked as answer]
The underlying [QTabPanel](https://quasar.dev/vue-components/tab-panels#qtabpanels-api) has a prop called `keep-alive`. With this line the views are not rebuild every time:

```py
with ui.tab_panels(tabs, value='A').props('keep-alive').classes('w-full'):
    ...
```

Note: the `w-full` class ensures that the width of the children are not affecting each other.
